R/init.R

```diff
@@ -34,11 +34,11 @@
 #'   The header of the app.
 #' @param footer (`shiny.tag` or `character(1)`)
 #'   The footer of the app.
-#' @param id (`character`)
-#'   module id to embed it, if provided,
-#'   the server function must be called with [shiny::moduleServer()];
-#'   See the vignette for an example. However, [ui_teal_with_splash()]
-#'   is then preferred to this function.
+#' @param id optional (`character`)
+#'   specifying the module id to be used when embedding the teal app within another `shiny` app as a module.
+#'   - If an id is provided, the server function must be called using [`shiny::moduleServer()`] with the specified id.
```

Comment:
I'm not sure if it is possible to call teal through `moduleServer`. Probably `callModule` would work instead. Have anyone tried to wrap teal output in other app?
 @averissimo 
 
```suggestion
#'   - If an id is provided, the server function must be called using [`shiny::moduleServer()`] with the specified id.
```

Reply:
It is, check out this "small" example

```r
teal_modules <- teal::modules(teal::example_module())

ui <- function(id) {
  ns <- shiny::NS(id)
  shiny::div(
    style = "background-color: #ffc6c4;",
    shiny::h1("This is a shiny app wrapper around teal"),
    shiny::hr(),
    shiny::div(
      style = "margin: 5em; border: 1px solid black; background-color: white",
      shiny::div(
        teal::ui_teal_with_splash(id = ns("teal_wrapper"), data = teal.data::teal_data())
      )
    )
  )
}

server <- function(id) {
  shiny::moduleServer(id, function(input, output, session) {
    teal::srv_teal_with_splash(
      id = "teal_wrapper", data = teal.data::teal_data(iris = iris), modules = teal_modules
    )
  })
}

shiny::shinyApp(ui("app"), function(input, output, session) server("app"))
```

![image](https://github.com/insightsengineering/teal/assets/211358/a6c9dc90-2403-467e-abfd-4764303cd9f8)